### PR TITLE
Allow trust builder

### PR DESCRIPTION
**Pull request by bartekn**
This PR adds AllowTrust operation mutators and default arm to return an error if mutator is not of expected type.

### CODE REVIEW

File: build/allow_trust_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package build
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stellar/go-stellar-base"
+	"github.com/stellar/go-stellar-base/xdr"
+)
+
+var _ = Describe("Payment Mutators", func() {
```
Comment (nullstyle) [inline build/allow_trust_test.go]: Looks like you didn't change the name here.

File: build/allow_trust_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+			It("sets the destination to the correct xdr.AccountId", func() {
+				aid, _ := stellarbase.AddressToAccountId(address)
+				Expect(subject.AT.Trustor.MustEd25519()).To(Equal(aid.MustEd25519()))
+			})
+		})
+
+		Context("using an invalid value", func() {
+			BeforeEach(func() { mut = Trustor{bad} })
+			It("failed", func() { Expect(subject.Err).To(HaveOccurred()) })
```
Comment (nullstyle) [inline build/allow_trust_test.go]: Tip:  There is a `MatchError` matcher in gomega that gives you a bit more specificity on errors that occur.  For example, you could write:

`Expect(subject.Err).To(MatchError("invalid trustor"))`

to match using a string against `subject.Err.Error()`, or you can do:

`Expect(subject.Err).To(MatchError(errors.New("some error"))` 

to match against a specific instance.

----

I'm not saying you should to change anything, just giving tips.

File: build/allow_trust_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			})
+		})
+
+		Context("using an invalid value", func() {
+			BeforeEach(func() { mut = SourceAccount{bad} })
+			It("failed", func() { Expect(subject.Err).To(HaveOccurred()) })
+		})
+	})
+
+	Describe("AllowTrustAsset", func() {
```
Comment (nullstyle) [inline build/allow_trust_test.go]: Should probably have a context for testing behavior when the code is over 12 bytes

File: build/allow_trust_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+			BeforeEach(func() {
+				mut = Authorize{true}
+			})
+
+			It("sets authorize flag properly", func() {
+				Expect(subject.AT.Authorize).To(Equal(true))
+			})
+		})
+
+		Context("when equal false", func() {
```
Comment (nullstyle) [inline build/allow_trust_test.go]: You might think about setting `subject.AT.Authorize` to `true` in your setup function, so you are confirming that the change is occurring.  The current form will always succeed because the zero value for `subject.AT.Authorize` is `false`.

**Comment by nullstyle** [comment]
+1, only a couple of minor cleanups needed, IMO

File: build/allow_trust_test.go (hunk 2 above)
Comment (bartekn) [inline build/allow_trust_test.go]: Good to know, thanks! :)

File: build/allow_trust_test.go (hunk 3 above)
Comment (bartekn) [inline build/allow_trust_test.go]: Right, I also added a test for empty string.
